src/ray/raylet/lineage_cache.cc

```diff
@@ -71,24 +70,22 @@ boost::optional<LineageEntry &> Lineage::GetEntryMutable(const UniqueID &task_id
   }
 }
 
-bool Lineage::SetEntry(LineageEntry &&new_entry) {
+bool Lineage::SetEntry(const Task &task, GcsStatus status) {
   // Get the status of the current entry at the key.
-  auto task_id = new_entry.GetEntryId();
-  GcsStatus current_status = GcsStatus::NONE;
-  auto current_entry = PopEntry(task_id);
+  auto task_id = task.GetTaskSpecification().TaskId();
+  auto current_entry = GetEntryMutable(task_id);
   if (current_entry) {
-    current_status = current_entry->GetStatus();
-  }
-
-  if (current_status < new_entry.GetStatus()) {
-    // If the new status is greater, then overwrite the current entry.
+    if (current_entry->GetStatus() < status) {
+      // If the new status is greater, then overwrite the current entry.
+      current_entry->SetStatus(status);
```

Comment:
`SetStatus` will return a `bool`, so you can do `if (current_entry->SetStatus(status))` if you'd like.